lib/rspec/core/formatters/base_formatter.rb

```diff
@@ -48,7 +48,7 @@ def example_group_started(notification)
         # @param _notification [NullNotification] (Ignored)
         # @see RSpec::Core::Formatters::Protocol#close
         def close(_notification)
-          restore_sync_output
+          restore_sync_output if output_supports_sync && !output.closed?
```

Comment:
This is a duplicate, its basically:
```
 output.sync = @old_sync if output_supports_sync && !output.closed? if output_supports_sync && !output.closed?
```
```suggestion
          restore_sync_output
```